### PR TITLE
Basic organizations api spaces

### PR DESCRIPTION
**Pull request by OAGr**

### CODE REVIEW

File: app/controllers/organizations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class OrganizationsController < ApplicationController
+  # TODO(matthew): Are there helpers for logged_in checking? Can we run a before_action here and elsewhere to strip the
+  # current_user checks?
```
Comment (OAGr) [inline app/controllers/organizations_controller.rb]: This definitely seems like you can use a before_action, maybe ``before_action :authorize_organization`` or something.  Maybe this routes to, ``user.can_access_organization(organization)``.  Not sure.

File: app/controllers/spaces_controller.rb [hunk 2]
```diff
@@ -5,11 +5,25 @@ class SpacesController < ApplicationController
 
   #GET /spaces
   #GET /spaces.json
+  #TODO(matthew): params['strings'] or params[:symbols]?
   def index
     if params['user_id']
```
Comment (OAGr) [inline app/controllers/spaces_controller.rb]: This should at least be split up into two other methods, maybe ``user_index`` and ``organization_index`` or ``user_spaces``.

File: app/controllers/spaces_controller.rb [hunk 3]
```diff
@@ -5,11 +5,25 @@ class SpacesController < ApplicationController
...
       @user = User.find(params['user_id'])
-      @spaces = @user.spaces.visible_by(current_user)
+      if current_user && @user.id == current_user.id
+        @spaces = @user.spaces
+      else
+        @spaces = @user.spaces.is_public
+      end
```
Comment (OAGr) [inline app/controllers/spaces_controller.rb]: why not `` @user.spaces.visible_by(current_user)``?

Comment (mmcdermott) [inline app/controllers/spaces_controller.rb]: It's not a scope anymore, its a class method to account for organizations. We could use a renamed scope here, but this way seemed cleaner to me anyways, less double scoping. Though maybe rails makes the sql the same anyhow.

File: app/controllers/spaces_controller.rb [hunk 4]
```diff
@@ -5,11 +5,25 @@ class SpacesController < ApplicationController
...
+    elsif params['organization_id']
+      if current_user && current_user.member_of?(params['organization_id'])
+        @spaces = Organization.find(params['organization_id']).spaces
+      else
+        head :unauthorized
```
Comment (OAGr) [inline app/controllers/spaces_controller.rb]: this section could be cleaned up (not for this PR).  I guess ``member_of?`` isn't very specific, but it's more low-level than I would want in a controller.

File: app/controllers/spaces_controller.rb [hunk 5]
```diff
@@ -5,11 +5,25 @@ class SpacesController < ApplicationController
...
+        return
+      end
     else
+      # TODO(matthew): Do we have a use for this case? It carries some extra baggage that I'd like to remove, if we
+      # don't really need it.
       @spaces = Space.visible_by(current_user).first(10)
```
Comment (OAGr) [inline app/controllers/spaces_controller.rb]: I don't think we need this.

File: app/controllers/spaces_controller.rb [hunk 6]
```diff
@@ -36,10 +50,13 @@ def create
 
     if !space_params.has_key? :is_private
       @space.is_private = @space.user.prefers_private?
+      if @space.organization
+        @space.is_private = @space.is_private || organization.prefers_private?
```
Comment (OAGr) [inline app/controllers/spaces_controller.rb]: are we allowing organizations to have public models?  If not, I imagine we should stay simple here.

File: app/models/space.rb [hunk 7]
```diff
@@ -16,7 +16,12 @@ class Space < ActiveRecord::Base
 
   scope :is_private, -> { where(is_private: true) }
   scope :is_public, -> { where(is_private: false) }
-  scope :visible_by, -> (user) { where 'is_private IS false OR user_id = ?', user.try(:id) }
+  scope :public_or_belonging_to, -> (user) { where 'is_private IS false OR user_id = ?', user.try(:id) }
+
+  def self.visible_by(user)
+    return public_or_belonging_to(user) if user.organization.nil?
```
Comment (OAGr) [inline app/models/space.rb]: Small comment, but this first line is a bit un-ruby like, if this is a typical case (that there's no organization).  These short statements are typically for catches / errors and stuff, while this seems like more important logic.

File: app/representers/spaces_representer.rb [hunk 8]
```diff
@@ -4,7 +4,7 @@ class SpacesRepresenter < Roar::Decorator
   include Roar::JSON
   include Roar::JSON::HAL
 
-  collection :to_a, as: "items", class: Space do
+  collection :to_a, as: 'items', class: Space do
```
Comment (OAGr) [inline app/representers/spaces_representer.rb]: good catch :)

File: config/routes.rb [hunk 9]
```diff
@@ -12,58 +12,8 @@
...
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  resources :organization, only: [:show]
+  resources :organizations do
+    resources :spaces, only: [:index]
```
Comment (OAGr) [inline config/routes.rb]: I think there's some way we could compress that above, but i realize it's like this elsewhere in the doc.

File: app/models/user.rb [hunk 10]
```diff
@@ -41,6 +41,7 @@ def can_create_private_models
   end
 
   def prefers_private?
+    # TODO(matthew): What happens when a free user joins a paid organization?
```
Comment (OAGr) [inline app/models/user.rb]: Users and organizations are totally different.

File: app/representers/space_representer.rb [hunk 11]
```diff
@@ -13,9 +13,12 @@ class SpaceRepresenter < Roar::Decorator
   property :is_private
   property :user_id
 
+  # TODO(matthew): These aren't being used atm. Should we remove them?
```
Comment (OAGr) [inline app/representers/space_representer.rb]: If you make sure they aren't actually being used, I feel like they are for some reason.  Can discuss.

**Pull request merged**